### PR TITLE
Don't overwrite the Content-Type when compressing

### PR DESCRIPTION
**Pull request by Hello71**
https://trac.torproject.org/projects/tor/ticket/28100

### CODE REVIEW

File: src/feature/dircache/dircache.c [hunk 1]
```diff
@@ -166,22 +166,16 @@ write_http_response_header_impl(dir_connection_t *conn, ssize_t length,
   buf_free(buf);
 }
 
-/** As write_http_response_header_impl, but sets encoding and content-typed
- * based on whether the response will be <b>compressed</b> or not. */
+/** As write_http_response_header_impl, but translates compress_method_t into
```
Comment (teor2345) [inline src/feature/dircache/dircache.c]: The argument name also changes from `method to `encoding`.
So let's say:
```suggestion
/** As write_http_response_header_impl, but translates method into
```

You can keep the types as well if you want.

File: src/feature/dircache/dircache.c [hunk 2]
```diff
@@ -166,22 +166,16 @@ write_http_response_header_impl(dir_connection_t *conn, ssize_t length,
...
+ * const char * */
```
Comment (teor2345) [inline src/feature/dircache/dircache.c]:
```suggestion
 * encoding */
```

**Comment by teor2345** [comment]
test-stem is now active on master's CI:
https://trac.torproject.org/projects/tor/ticket/27913

Closing and re-opening to re-run the merged branch.

**Comment by nmathewson** [comment]
squashed and merged

**Pull request closed without merging**